components/QrScanner.tsx

```diff
@@ -43,21 +43,34 @@ export const QrScanner: React.FC<QrScannerProps> = (props) => {
     return <View />;
   }
 
-  if (hasPermission === false) {
+  const CameraDisabledPopUp: React.FC = () => {
     return (
-      <Column padding="24" fill align="space-between">
-        <Centered fill>
-          <Text align="center" color={Theme.Colors.errorMessage}>
-            {t('missingPermissionText')}
-          </Text>
-        </Centered>
-        <Button title={t('allowCameraButton')} onPress={openSettings}></Button>
-      </Column>
+      <View style={{ display: hasPermission ? 'none' : 'flex' }}>
+        <Row style={Theme.Styles.cameraDisabledPopUp}>
+          <Column>
+            <Text color={Theme.Colors.whiteText} weight="bold">
+              {t('cameraAccessDisabled')}
+            </Text>
+            <Text
+              color={Theme.Colors.whiteText}
+              weight="semibold"
+              size="smaller">
+              {t('cameraPermissionGuideLabel')}
+            </Text>
+          </Column>
+          <Icon
+            name="close"
+            onPress={openSettings}
+            color={Theme.Colors.whiteText}
+            size={19}
+          />
+        </Row>
+      </View>
     );
-  }
-
+  };
   return (
     <View>
+      <CameraDisabledPopUp />
```

Comment:
we can render this <CameraDisabledPopUp /> component, only when we don't have permission instead of always rendering this component.

we can use conditonal rendering to toggle

Reply:
I have already rendered it by using conditional rendering in the above code. Also It working fine as per conditions.

Reply:
hasPermission field only toggles the display of the component. although this will get rendered all the time. since we want to render the component only when hasPermission is false, so its better to use conditional rendering like this.

{hasPermission && <Component />}